src/dmd/dclass.d

```diff
@@ -438,22 +438,13 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
     /*******************************************
      * Determine if 'this' is a base class of cd.
      */
-    bool isBaseOf(ClassDeclaration cd, int* poffset)
+    bool isBaseOf(ClassDeclaration cd, int* poffset) pure nothrow @nogc
     {
         //printf("ClassDeclaration.isBaseOf(this = '%s', cd = '%s')\n", toChars(), cd.toChars());
         if (poffset)
             *poffset = 0;
         while (cd)
         {
-            /* cd.baseClass might not be set if cd is forward referenced.
-             */
-            if (!cd.baseClass && cd.semanticRun < PASS.semanticdone && !cd.isInterfaceDeclaration())
-            {
-                cd.dsymbolSemantic(null);
```

Comment:
Is it worth replacing this check/call with an assert to catch potential regressions that pass forward references?